KLSadd_insertion/src/updateChapters.py

```diff
@@ -187,162 +468,197 @@ def referencePlacer(chap, references, p, kls, refII, refIII, specref):
-            indexes.append(index - 1)
-        if ("paragraph{" in word) and (index > 313):
-            klsparas.append(index - 1)
-    print(indexes)
-    print(specref9)
-    print(specref14)
-    print(klsparas)
-    print(mathPeople)
-    # now indexes holds all of the places there is a section
-    # using these indexes, get all of the words in between and add that to the
-    # paras[]
-    for i in range(len(indexes) - 1):
-        box = ''.join(addendum[indexes[i]: indexes[i + 1] - 1])
-        paras.append(box)
-    box2 = ''.join(addendum[indexes[35]: 2245])
-    paras.append(box2)
-    # paras now holds the paragraphs that need to go into the chapter files, but they need to go in the appropriate
-    # section(like Wilson, Racah, Hahn, etc.) so we use the mathPeople variable
-    # we can use the section names to place the relevant paragraphs in the
-    # right place
-
-    # as of 2/8/16 the paragraphs will go before the References paragraph of the relevant section
-    # parse both files 9 and 14 as strings
-
-    # chapter 9
-    with open("chap09.tex", "r") as ch9:
-        entire9 = ch9.readlines()  # reads in as a list of strings
-    ch9.close()
-
-    # chapter 14
-    with open("chap14.tex", "r") as ch14:
-        entire14 = ch14.readlines()
-    ch14.close()
-    # call the findReferences method to find the index of the References paragraph in the two file strings
-    # two variables for the references lists one for chapter 9 one for chapter
-    # 14
-    chapticker = 0
-    references9 = findReferences(entire9)
-    chapticker += 1
-    references14 = findReferences(entire14)
-    # call the fixChapter method to get a list with the addendum paragraphs
-    # added in
-    chapticker2 = 0
-    str9 = ''.join(
-        fixChapter(
-            entire9,
-            references9,
-            paras,
-            addendum,
-            ref9II,
-            ref9III,
-            specref9))
-    chapticker2 += 1
-    str14 = ''.join(
-        fixChapter(
-            entire14,
-            references14,
-            paras,
-            addendum,
-            ref14II,
-            ref14III,
-            specref14))
-
-"""
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-If you are writing something that will make a change to the chapter files, write it BEFORE this line, this part
-is where the lists representing the words/strings in the chapter are joined together and updated as a string!
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""
-
-
-# write to files
-# new output files for safety
-with open("updated9.tex", "w") as temp9:
-    temp9.write(str9)
-
-with open("updated14.tex", "w") as temp14:
-    temp14.write(str14)
+
+def main():
+    """
+    Runs all of the other functions and outputs their results into an output file as well as putting together the list
+    of additions that is fed into fix_chapter.
+
+    :return:
```

Comment:
as @notjagan said, if the function is actually not returning anything, remove the :return: stuff